### DISCUSSION TITLE
Help setting up NFS access on Synology NAS and Debian Virtual Machine (Running on Synology NAS)

### DISCUSSION THREAD

**Discussion by drinksomewhiskey** [retronas/discussions#22] [Q&A]
Hi everyone,

I have searched and searched, read through every post containing "NFS" in them, but am stuck and have had no luck getting this to work.

The quick summary is that I have a Synology DS1019+ NAS, where I spun up a Debian VM and successfully installed both RetroNAS and Cockpit. I am able to connect the NFS share located on my Synology NAS within Cockpit, set the appropriate top level directory in RetroNAS however, I seem to always get what it seems like permission errors anytime I try to install any services.

The longer story is the following:

_**Setting up RetroNAS**_
I successfully spun up a Debian Bullseye VM running on the Synology DS1019+ and installed RetroNAS and Cockpit accordingly (following Dan's instructional videos on YouTube).

_**Setting up NFS on Synology DS1019+**_
I then set up the NFS permissions on the Synology DS1019+ by going into the settings of the directory I want to enable NFS on and selected the following:

- Hostname or IP: 192.168.1.142 (this is the IP to the Debian VM running on the Synology DS1019+)
- Privilege: Read/Write
- Squash: Map root to admin
- Security: sys

I also enabled asynchronous, allow connections from non-privileged ports (ports higher than 1024), and allow users to access mounted subfolders.

_Here is an image with my settings within my Synology DS1019+:_
<img width="849" alt="Screenshot 2023-02-01 at 9 49 09 PM" src="https://user-images.githubusercontent.com/122244876/216242524-ba555aa2-f1bf-46f8-b7f9-fdb7a82dd28f.png">

_**Mounting NFS on Cockpit**_
I then mounted the NFS on Cockpit using following:

- Server address: 192.168.1.57 (IP of the Synology DS1019+)
- Path on server: volume1/Video Games (this is the path on the Synology DS1019+)
- Local mount point: /storage

I have also enabled mount at boot.

_Here are images with my settings within Cockpit:_
<img width="730" alt="Screenshot 2023-02-01 at 9 55 39 PM" src="https://user-images.githubusercontent.com/122244876/216245208-501e7e95-d977-4fe7-af5b-8eee98a49539.png">

<img width="1740" alt="Screenshot 2023-02-01 at 10 06 54 PM" src="https://user-images.githubusercontent.com/122244876/216245259-f5ddd28f-c0b3-4bf9-b42d-4b4bd32a5e8b.png">

_**RetroNAS settings**_
Lastly I set the top level directory to be /storage under global settings.

_Here is an image with my settings:_
<img width="440" alt="Screenshot 2023-02-01 at 9 56 33 PM" src="https://user-images.githubusercontent.com/122244876/216245281-191eb6d0-e9ce-4788-a49e-07ba78d2bdc0.png">

I also set up a user and password within RetroNAS using the same username and password I used for the Debian VM.

_**Errors when installing services on RetroNAS**_
Finally, when trying to install any services, I get the following errors:

_Here is an image showing the errors when trying to install Samba:_
<img width="2056" alt="Screenshot 2023-02-01 at 10 05 41 PM" src="https://user-images.githubusercontent.com/122244876/216245368-3396c75b-4442-445e-bc13-2f976ac24bde.png">

I get similar errors from trying to install other services as well - nothing installs successfully.

I have tried many different potential solutions, also read through the RetroNAS subreddit, but have had no luck. I only found another post on reddit stating similar issues on a Synology NAS, where the person just gave up and reverted to a USB drive. Does anyone see any issues on my setup from above to help me solve this? Any tips would help!

Thank you in advance!

**Comment by HylianPeasant** [comment]
Hello,

I don't have specific advice to give.

Try to check if uid of the user you created on the nas match the uid of the user in your vm.
If not, trying changing the uid on the vm side (best that way) to match to one of the owner on the nas side.

NFS doesn't rely on username and password however, don't bother with that for your troubleshoot.

**Comment by HylianPeasant** [comment]
For changing uid/gid, you can follow this guide : https://muffinresearch.co.uk/linux-changing-uids-and-gids-for-user/
This is better using another account to change uid/gid of the account you want to update.

**Reply by drinksomewhiskey** [reply]
I really appreciate your help. I was having issues at first until I realized to do it via a secondary user. 

Also, I responded separately in your thread on this topic but wanted to reply here in case anyone else has issues. Aside from the users not matching, my other issue was the user mapping on the Synology which I set to do not map. 

Just FYI for anyone reading this. 

Again thank you for your help.
